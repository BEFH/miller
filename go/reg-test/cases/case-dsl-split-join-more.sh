run_mlr --from $indir/abixy-het put -q 'o = joink($*, ":"); print o'
run_mlr --from $indir/abixy-het put -q 'o = joinv($*, ":"); print o'
run_mlr --from $indir/abixy-het put -q 'o = joinkv($*, ":", ";"); print o'

run_mlr --from $indir/abixy-het put -q 'o = joink({1:2,  "abc":4, 5:"xyz"}, ":"); print o'
run_mlr --from $indir/abixy-het put -q 'o = joinv({1:2,  "abc":4, 5:"xyz"}, ":"); print o'
run_mlr --from $indir/abixy-het put -q 'o = joinkv({1:2, "abc":4, 5:"xyz"}, ":", ";"); print o'

run_mlr --from $indir/abixy-het put '$* = splitnv("a,b,c" , ","); for (k,v in $*) { print k.":".typeof(k)." ".v.":".typeof(v)}'
run_mlr --from $indir/abixy-het put '$* = splitkv("a=1,b=2,c", "=", ","); for (k, v in $*) {print k.":".typeof(k)." ".v.":".typeof(v)}'
run_mlr --from $indir/abixy-het put '$* = splitnv("a,b,c", IFS); print ">>".IRS."<<"; for (k, v in $*) {print k.":".typeof(k)." ".v.":".typeof(v)}'
run_mlr --from $indir/abixy-het put '$* = splitkv("a=1,b=2,c", IPS, IFS); print ">>".IRS."<<"; for (k, v in $*) {print k.":".typeof(k)." ".v.":".typeof(v)}'
run_mlr --from $indir/abixy-het put '$* = splitnv("a,b,c", OFS); print ">>".ORS."<<"; for (k, v in $*) {print k.":".typeof(k)." ".v.":".typeof(v)}'
run_mlr --from $indir/abixy-het put '$* = splitkv("a=1,b=2,c", OPS, OFS); print ">>".ORS."<<"; for (k, v in $*) {print k.":".typeof(k)." ".v.":".typeof(v)}'

run_mlr --from $indir/abixy-het put '$* = splitnvx("a,b,c" , ","); for (k, v in $*) {print k.":".typeof(k)." ".v.":".typeof(v)}'
run_mlr --from $indir/abixy-het put '$* = splitkvx("a=1,b=2,c", "=", ","); for (k, v in $*) {print k.":".typeof(k)." ".v.":".typeof(v)}'
run_mlr --from $indir/abixy-het put '$* = splitnvx("a,b,c", IFS); print ">>".IRS."<<"; for (k, v in $*) {print k.":".typeof(k)." ".v.":".typeof(v)}'
run_mlr --from $indir/abixy-het put '$* = splitkvx("a=1,b=2,c", IPS, IFS); print ">>".IRS."<<"; for (k, v in $*) {print k.":".typeof(k)." ".v.":".typeof(v)}'
run_mlr --from $indir/abixy-het put '$* = splitnvx("a,b,c", OFS); print ">>".ORS."<<"; for (k, v in $*) {print k.":".typeof(k)." ".v.":".typeof(v)}'
run_mlr --from $indir/abixy-het put '$* = splitkvx("a=1,b=2,c", OPS, OFS); print ">>".ORS."<<"; for (k, v in $*) {print k.":".typeof(k)." ".v.":".typeof(v)}'

run_mlr --oxtab --from $indir/abixy-het put 's = joink($*, ":"); $* = splitnv(s, ":")'
run_mlr --oxtab --from $indir/abixy-het put 's = joinv($*, ":"); $* = splitnv(s, ":")'
run_mlr --oxtab --from $indir/abixy-het put 's = joinkv($*, ":", ";"); $* = splitkv(s, ":", ";"); for (k,v in $*) { print k.":".typeof(k)." ".v.":".typeof(v)}'
run_mlr --oxtab --from $indir/abixy-het put 's = joinkv($*, ":", ";"); $* = splitkvx(s, ":", ";"); for (k,v in $*) { print k.":".typeof(k)." ".v.":".typeof(v)}'

run_mlr --oxtab --from $indir/abixy-het put 's = joink({1:2, "abc":4, 5:"xyz"}, ":"); $* = splitnv(s, ":")'
run_mlr --oxtab --from $indir/abixy-het put 's = joinv({1:2, "abc":4, 5:"xyz"}, ":"); $* = splitnv(s, ":")'
run_mlr --oxtab --from $indir/abixy-het put 's = joinkv({1:2, "abc":4, 5:"xyz"}, ":", ";"); $* = splitkv(s, ":", ";"); for (k,v in $*) { print k.":".typeof(k)." ".v.":".typeof(v)}'
run_mlr --oxtab --from $indir/abixy-het put 's = joinkv({1:2, "abc":4, 5:"xyz"}, ":", ";"); $* = splitkvx(s, ":", ";"); for (k,v in $*) { print k.":".typeof(k)." ".v.":".typeof(v)}'
