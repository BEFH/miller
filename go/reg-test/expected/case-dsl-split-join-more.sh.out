
mlr --from ./reg-test/input/abixy-het put -q o = joink($*, ":"); print o
a:b:i:x:y
a:b:i:x:y
aaa:b:i:x:y
a:bbb:i:x:y
a:b:i:xxx:y
a:b:i:x:y
a:b:iii:x:y
a:b:i:x:yyy
aaa:bbb:i:x:y
a:b:i:x:y

mlr --from ./reg-test/input/abixy-het put -q o = joinv($*, ":"); print o
pan:pan:1:0.3467901443380824:0.7268028627434533
eks:pan:2:0.7586799647899636:0.5221511083334797
wye:wye:3:0.20460330576630303:0.33831852551664776
eks:wye:4:0.38139939387114097:0.13418874328430463
wye:pan:5:0.5732889198020006:0.8636244699032729
zee:pan:6:0.5271261600918548:0.49322128674835697
eks:zee:7:0.6117840605678454:0.1878849191181694
zee:wye:8:0.5985540091064224:0.976181385699006
hat:wye:9:0.03144187646093577:0.7495507603507059
pan:wye:10:0.5026260055412137:0.9526183602969864

mlr --from ./reg-test/input/abixy-het put -q o = joinkv($*, ":", ";"); print o
a:pan;b:pan;i:1;x:0.3467901443380824;y:0.7268028627434533
a:eks;b:pan;i:2;x:0.7586799647899636;y:0.5221511083334797
aaa:wye;b:wye;i:3;x:0.20460330576630303;y:0.33831852551664776
a:eks;bbb:wye;i:4;x:0.38139939387114097;y:0.13418874328430463
a:wye;b:pan;i:5;xxx:0.5732889198020006;y:0.8636244699032729
a:zee;b:pan;i:6;x:0.5271261600918548;y:0.49322128674835697
a:eks;b:zee;iii:7;x:0.6117840605678454;y:0.1878849191181694
a:zee;b:wye;i:8;x:0.5985540091064224;yyy:0.976181385699006
aaa:hat;bbb:wye;i:9;x:0.03144187646093577;y:0.7495507603507059
a:pan;b:wye;i:10;x:0.5026260055412137;y:0.9526183602969864

mlr --from ./reg-test/input/abixy-het put -q o = joink({1:2,  "abc":4, 5:"xyz"}, ":"); print o
1:abc:5
1:abc:5
1:abc:5
1:abc:5
1:abc:5
1:abc:5
1:abc:5
1:abc:5
1:abc:5
1:abc:5

mlr --from ./reg-test/input/abixy-het put -q o = joinv({1:2,  "abc":4, 5:"xyz"}, ":"); print o
2:4:xyz
2:4:xyz
2:4:xyz
2:4:xyz
2:4:xyz
2:4:xyz
2:4:xyz
2:4:xyz
2:4:xyz
2:4:xyz

mlr --from ./reg-test/input/abixy-het put -q o = joinkv({1:2, "abc":4, 5:"xyz"}, ":", ";"); print o
1:2;abc:4;5:xyz
1:2;abc:4;5:xyz
1:2;abc:4;5:xyz
1:2;abc:4;5:xyz
1:2;abc:4;5:xyz
1:2;abc:4;5:xyz
1:2;abc:4;5:xyz
1:2;abc:4;5:xyz
1:2;abc:4;5:xyz
1:2;abc:4;5:xyz

mlr --from ./reg-test/input/abixy-het put $* = splitnv("a,b,c" , ","); for (k,v in $*) { print k.":".typeof(k)." ".v.":".typeof(v)}
1:string a:string
2:string b:string
3:string c:string
1=a,2=b,3=c
1:string a:string
2:string b:string
3:string c:string
1:string a:string
2:string b:string
3:string c:string
1:string a:string
2:string b:string
3:string c:string
1:string a:string
2:string b:string
3:string c:string
1=a,2=b,3=c
1=a,2=b,3=c
1=a,2=b,3=c
1=a,2=b,3=c
1:string a:string
2:string b:string
3:string c:string
1=a,2=b,3=c
1:string a:string
2:string b:string
3:string c:string
1=a,2=b,3=c
1:string a:string
2:string b:string
3:string c:string
1=a,2=b,3=c
1:string a:string
2:string b:string
3:string c:string
1:string a:string
1=a,2=b,3=c
2:string b:string
3:string c:string
1=a,2=b,3=c

mlr --from ./reg-test/input/abixy-het put $* = splitkv("a=1,b=2,c", "=", ","); for (k, v in $*) {print k.":".typeof(k)." ".v.":".typeof(v)}
a:string 1:int
b:string 2:int
3:string c:string
a=1,b=2,3=c
a:string 1:int
b:string 2:int
3:string c:string
a:string 1:int
b:string 2:int
3:string c:string
a=1,b=2,3=c
a:string 1:int
b:string 2:int
a=1,b=2,3=c
3:string c:string
a:string 1:int
b:string 2:int
3:string c:string
a=1,b=2,3=c
a=1,b=2,3=c
a:string 1:int
b:string 2:int
3:string c:string
a=1,b=2,3=c
a:string 1:int
b:string 2:int
3:string c:string
a=1,b=2,3=c
a:string 1:int
b:string 2:int
3:string c:string
a:string 1:int
b:string 2:int
3:string c:string
a=1,b=2,3=c
a=1,b=2,3=c
a:string 1:int
b:string 2:int
3:string c:string
a=1,b=2,3=c

mlr --from ./reg-test/input/abixy-het put $* = splitnv("a,b,c", IFS); print ">>".IRS."<<"; for (k, v in $*) {print k.":".typeof(k)." ".v.":".typeof(v)}
>>
<<
1:string a:string
2:string b:string
3:string c:string
>>
<<1=a,2=b,3=c

1:string a:string
2:string b:string
3:string c:string
>>
<<
1:string a:string
2:string b:string
3:string c:string
1=a,2=b,3=c
1=a,2=b,3=c
>>
<<
1:string a:string
2:string b:string
3:string c:string
>>
<<
1:string a:string1=a,2=b,3=c

2:string b:string
3:string c:string
>>
<<
1=a,2=b,3=c
1:string a:string
2:string b:string
3:string c:string
>>
<<
1:string a:string
2:string b:string
1=a,2=b,3=c
3:string c:string
1=a,2=b,3=c
>>
<<
1:string a:string
2:string b:string
3:string c:string
>>
<<
1:string a:string
1=a,2=b,3=c
2:string b:string
3:string c:string
>>
<<
1:string a:string
1=a,2=b,3=c
2:string b:string
3:string c:string
1=a,2=b,3=c

mlr --from ./reg-test/input/abixy-het put $* = splitkv("a=1,b=2,c", IPS, IFS); print ">>".IRS."<<"; for (k, v in $*) {print k.":".typeof(k)." ".v.":".typeof(v)}
>>
<<
a:string 1:int
b:string 2:int
3:string c:string
>>
<<
a=1,b=2,3=c
a:string 1:int
b:string 2:int
3:string c:string
>>
<<
a:string 1:int
b:string 2:int
3:string c:stringa=1,b=2,3=c

>>
<<
a:string 1:int
b:string 2:int
3:string c:string
a=1,b=2,3=c
a=1,b=2,3=c
>>
<<
a:string 1:int
b:string 2:int
3:string c:string
>>
<<
a:string 1:int
b:string 2:int
a=1,b=2,3=c
3:string c:string
>>
<<
a:string 1:int
b:string 2:int
3:string c:string
a=1,b=2,3=c
a=1,b=2,3=c
>>
<<
a:string 1:int
b:string 2:int
3:string c:string
a=1,b=2,3=c
>>
<<
a:string 1:int
b:string 2:int
3:string c:string
a=1,b=2,3=c
>>
<<
a:string 1:int
b:string 2:int
3:string c:string
a=1,b=2,3=c

mlr --from ./reg-test/input/abixy-het put $* = splitnv("a,b,c", OFS); print ">>".ORS."<<"; for (k, v in $*) {print k.":".typeof(k)." ".v.":".typeof(v)}
>>
<<
1:string a:string
2:string b:string
3:string c:string
>>
<<
1:string a:string
2:string b:string1=a,2=b,3=c

3:string c:string
1=a,2=b,3=c
>>
<<
1:string a:string
2:string b:string
3:string c:string
>>
<<
1:string a:string
2:string b:string
1=a,2=b,3=c
3:string c:string
>>
<<
1:string a:string
2:string b:string
3:string c:string
1=a,2=b,3=c
1=a,2=b,3=c
>>
<<
1:string a:string
2:string b:string
3:string c:string
>>
<<
1:string a:string
2:string b:string
1=a,2=b,3=c
3:string c:string
1=a,2=b,3=c
>>
<<
1:string a:string
2:string b:string
3:string c:string
>>
<<
1:string a:string
2:string b:string
3:string c:string
1=a,2=b,3=c
1=a,2=b,3=c
>>
<<
1:string a:string
2:string b:string
3:string c:string
1=a,2=b,3=c

mlr --from ./reg-test/input/abixy-het put $* = splitkv("a=1,b=2,c", OPS, OFS); print ">>".ORS."<<"; for (k, v in $*) {print k.":".typeof(k)." ".v.":".typeof(v)}
>>
<<
a:string 1:int
b:string 2:int
3:string c:string
>>
<<
a=1,b=2,3=c
a:string 1:int
b:string 2:int
3:string c:string
>>
<<
a=1,b=2,3=c
a:string 1:int
b:string 2:int
3:string c:string
>>
<<
a=1,b=2,3=c
a:string 1:int
b:string 2:int
3:string c:string
>>
<<
a:string 1:int
b:string 2:int
3:string c:string
a=1,b=2,3=c
a=1,b=2,3=c
>>
<<
a:string 1:int
b:string 2:int
3:string c:string
a=1,b=2,3=c
>>
<<
a:string 1:int
b:string 2:int
3:string c:string
>>
<<
a:string 1:int
b:string 2:int
3:string c:stringa=1,b=2,3=c

>>
<<
a:string 1:int
a=1,b=2,3=c
b:string 2:int
3:string c:string
a=1,b=2,3=c
>>
<<
a:string 1:int
b:string 2:int
3:string c:string
a=1,b=2,3=c

mlr --from ./reg-test/input/abixy-het put $* = splitnvx("a,b,c" , ","); for (k, v in $*) {print k.":".typeof(k)." ".v.":".typeof(v)}
1:string a:string
2:string b:string
3:string c:string
1=a,2=b,3=c
1:string a:string
2:string b:string
3:string c:string
1:string a:string
2:string b:string
3:string c:string
1=a,2=b,3=c
1=a,2=b,3=c
1:string a:string
2:string b:string
3:string c:string
1=a,2=b,3=c
1:string a:string
2:string b:string
3:string c:string
1:string a:string
2:string b:string
3:string c:string
1=a,2=b,3=c
1=a,2=b,3=c
1:string a:string
2:string b:string
3:string c:string
1:string a:string
2:string b:string1=a,2=b,3=c

3:string c:string
1:string a:string
2:string b:string
3:string c:string
1=a,2=b,3=c
1=a,2=b,3=c
1:string a:string
2:string b:string
3:string c:string
1=a,2=b,3=c

mlr --from ./reg-test/input/abixy-het put $* = splitkvx("a=1,b=2,c", "=", ","); for (k, v in $*) {print k.":".typeof(k)." ".v.":".typeof(v)}
a:string 1:string
b:string 2:string
3:string c:string
a=1,b=2,3=c
a:string 1:string
b:string 2:string
3:string c:string
a=1,b=2,3=c
a:string 1:string
b:string 2:string
3:string c:string
a:string 1:string
b:string 2:string
3:string c:string
a:string 1:string
b:string 2:string
a=1,b=2,3=c
a=1,b=2,3=c
3:string c:string
a=1,b=2,3=c
a:string 1:string
b:string 2:string
3:string c:string
a:string 1:string
b:string 2:string
3:string c:string
a=1,b=2,3=c
a=1,b=2,3=c
a:string 1:string
b:string 2:string
3:string c:string
a=1,b=2,3=c
a:string 1:string
b:string 2:string
3:string c:string
a:string 1:string
b:string 2:string
3:string c:string
a=1,b=2,3=c
a=1,b=2,3=c

mlr --from ./reg-test/input/abixy-het put $* = splitnvx("a,b,c", IFS); print ">>".IRS."<<"; for (k, v in $*) {print k.":".typeof(k)." ".v.":".typeof(v)}
>>
<<
1:string a:string
2:string b:string
3:string c:string
>>
<<
1=a,2=b,3=c
1:string a:string
2:string b:string
3:string c:string
>>
<<
1=a,2=b,3=c
1:string a:string
2:string b:string
3:string c:string
>>
<<
1=a,2=b,3=c
1:string a:string
2:string b:string
3:string c:string
1=a,2=b,3=c
>>
<<
1:string a:string
2:string b:string
3:string c:string
>>
<<
1:string a:string
1=a,2=b,3=c
2:string b:string
3:string c:string
>>
<<
1:string a:string
2:string b:string
3:string c:string
1=a,2=b,3=c
1=a,2=b,3=c
>>
<<
1:string a:string
2:string b:string
3:string c:string
1=a,2=b,3=c
>>
<<
1:string a:string
2:string b:string
3:string c:string
>>
<<
1:string a:string
2:string b:string
3:string c:string
1=a,2=b,3=c
1=a,2=b,3=c

mlr --from ./reg-test/input/abixy-het put $* = splitkvx("a=1,b=2,c", IPS, IFS); print ">>".IRS."<<"; for (k, v in $*) {print k.":".typeof(k)." ".v.":".typeof(v)}
>>
<<
a:string 1:string
b:string 2:string
3:string c:string
>>
<<a=1,b=2,3=c

a:string 1:string
b:string 2:string
3:string c:string
>>
<<
a=1,b=2,3=c
a:string 1:string
b:string 2:string
3:string c:string
a=1,b=2,3=c
>>
<<
a:string 1:string
b:string 2:string
3:string c:string
>>
<<
a:string 1:string
b:string 2:string
a=1,b=2,3=c
3:string c:string
>>
<<
a:string 1:string
b:string 2:string
3:string c:string
>>
<<
a:string 1:string
b:string 2:string
3:string c:string
a=1,b=2,3=c
a=1,b=2,3=c
a=1,b=2,3=c
>>
<<
a:string 1:string
b:string 2:string
3:string c:string
>>
<<
a=1,b=2,3=c
a:string 1:string
b:string 2:string
3:string c:string
>>
<<
a:string 1:string
b:string 2:string
3:string c:string
a=1,b=2,3=c
a=1,b=2,3=c

mlr --from ./reg-test/input/abixy-het put $* = splitnvx("a,b,c", OFS); print ">>".ORS."<<"; for (k, v in $*) {print k.":".typeof(k)." ".v.":".typeof(v)}
>>
<<
1:string a:string
2:string b:string
3:string c:string
>>
<<
1=a,2=b,3=c
1:string a:string
2:string b:string
3:string c:string
>>
<<
1=a,2=b,3=c
1:string a:string
2:string b:string
3:string c:string
>>
<<
1:string a:string
2:string b:string
3:string c:string
1=a,2=b,3=c
1=a,2=b,3=c
>>
<<
1:string a:string
2:string b:string
3:string c:string
>>
<<
1:string a:string
2:string b:string
3:string c:string
>>
<<
1:string a:string
1=a,2=b,3=c
1=a,2=b,3=c
2:string b:string
3:string c:string
>>
<<
1:string a:string
2:string b:string
3:string c:string
1=a,2=b,3=c
>>
<<1=a,2=b,3=c

1:string a:string
2:string b:string
3:string c:string
>>
<<
1:string a:string
2:string b:string
3:string c:string
1=a,2=b,3=c
1=a,2=b,3=c

mlr --from ./reg-test/input/abixy-het put $* = splitkvx("a=1,b=2,c", OPS, OFS); print ">>".ORS."<<"; for (k, v in $*) {print k.":".typeof(k)." ".v.":".typeof(v)}
>>
<<
a:string 1:string
b:string 2:string
3:string c:string
>>
<<a=1,b=2,3=c

a:string 1:string
b:string 2:string
3:string c:string
>>
<<
a:string 1:string
b:string 2:string
3:string c:string
a=1,b=2,3=c
a=1,b=2,3=c
>>
<<
a:string 1:string
b:string 2:string
3:string c:string
>>
<<
a:string 1:string
b:string 2:string
3:string c:string
>>
<<
a:string 1:string
b:string 2:string
3:string c:string
>>
<<
a:string 1:string
b:string 2:string
3:string c:string
>>
<<
a:string 1:string
b:string 2:string
3:string c:string
a=1,b=2,3=c
a=1,b=2,3=c
a=1,b=2,3=c
a=1,b=2,3=c
a=1,b=2,3=c
>>
<<
a:string 1:string
b:string 2:string
3:string c:string
a=1,b=2,3=c
>>
<<
a:string 1:string
b:string 2:string
3:string c:string
a=1,b=2,3=c

mlr --oxtab --from ./reg-test/input/abixy-het put s = joink($*, ":"); $* = splitnv(s, ":")
1 a
2 b
3 i
4 x
5 y

1 a
2 b
3 i
4 x
5 y

1 aaa
2 b
3 i
4 x
5 y

1 a
2 bbb
3 i
4 x
5 y

1 a
2 b
3 i
4 xxx
5 y

1 a
2 b
3 i
4 x
5 y

1 a
2 b
3 iii
4 x
5 y

1 a
2 b
3 i
4 x
5 yyy

1 aaa
2 bbb
3 i
4 x
5 y

1 a
2 b
3 i
4 x
5 y

mlr --oxtab --from ./reg-test/input/abixy-het put s = joinv($*, ":"); $* = splitnv(s, ":")
1 pan
2 pan
3 1
4 0.3467901443380824
5 0.7268028627434533

1 eks
2 pan
3 2
4 0.7586799647899636
5 0.5221511083334797

1 wye
2 wye
3 3
4 0.20460330576630303
5 0.33831852551664776

1 eks
2 wye
3 4
4 0.38139939387114097
5 0.13418874328430463

1 wye
2 pan
3 5
4 0.5732889198020006
5 0.8636244699032729

1 zee
2 pan
3 6
4 0.5271261600918548
5 0.49322128674835697

1 eks
2 zee
3 7
4 0.6117840605678454
5 0.1878849191181694

1 zee
2 wye
3 8
4 0.5985540091064224
5 0.976181385699006

1 hat
2 wye
3 9
4 0.03144187646093577
5 0.7495507603507059

1 pan
2 wye
3 10
4 0.5026260055412137
5 0.9526183602969864

mlr --oxtab --from ./reg-test/input/abixy-het put s = joinkv($*, ":", ";"); $* = splitkv(s, ":", ";"); for (k,v in $*) { print k.":".typeof(k)." ".v.":".typeof(v)}
a:string pan:string
b:string pan:string
i:string 1:int
x:string 0.3467901443380824:float
y:string 0.7268028627434533:float
a pan
b pan
i 1
x 0.3467901443380824
y 0.7268028627434533
a:string eks:string
b:string pan:string
i:string 2:int
x:string 0.7586799647899636:float
y:string 0.5221511083334797:float

a eks
b pan
i 2
x 0.7586799647899636
y 0.5221511083334797
aaa:string wye:string
b:string wye:string
i:string 3:int
x:string 0.20460330576630303:float
y:string 0.33831852551664776:float
a:string eks:string
bbb:string wye:string

aaa wye
b   wye
i   3
x   0.20460330576630303
y   0.33831852551664776
i:string 4:int
x:string 0.38139939387114097:float
y:string 0.13418874328430463:float
a:string wye:string
b:string pan:string
i:string 5:int
xxx:string 0.5732889198020006:float

a   eks
bbb wye
i   4
x   0.38139939387114097
y   0.13418874328430463
y:string 0.8636244699032729:float

a   wye
b   pan
i   5
xxx 0.5732889198020006
y   0.8636244699032729
a:string zee:string
b:string pan:string
i:string 6:int
x:string 0.5271261600918548:float
y:string 0.49322128674835697:float

a zee
b pan
i 6
x 0.5271261600918548
y 0.49322128674835697
a:string eks:string
b:string zee:string
iii:string 7:int
x:string 0.6117840605678454:float
y:string 0.1878849191181694:float

a   eks
b   zee
iii 7
x   0.6117840605678454
y   0.1878849191181694
a:string zee:string
b:string wye:string
i:string 8:int
x:string 0.5985540091064224:float
yyy:string 0.976181385699006:float

a   zee
b   wye
i   8
x   0.5985540091064224
yyy 0.976181385699006
aaa:string hat:string
bbb:string wye:string
i:string 9:int
x:string 0.03144187646093577:float
y:string 0.7495507603507059:float

aaa hat
bbb wye
i   9
x   0.03144187646093577
y   0.7495507603507059
a:string pan:string
b:string wye:string
i:string 10:int
x:string 0.5026260055412137:float
y:string 0.9526183602969864:float

a pan
b wye
i 10
x 0.5026260055412137
y 0.9526183602969864

mlr --oxtab --from ./reg-test/input/abixy-het put s = joinkv($*, ":", ";"); $* = splitkvx(s, ":", ";"); for (k,v in $*) { print k.":".typeof(k)." ".v.":".typeof(v)}
a:string pan:string
b:string pan:string
i:string 1:string
x:string 0.3467901443380824:string
y:string 0.7268028627434533:string
a:string eks:string
b:string pan:string
i:string 2:string
x:string 0.7586799647899636:string
y:string 0.5221511083334797:string
a pan
b pan
i 1
x 0.3467901443380824
y 0.7268028627434533

a eks
b pan
i 2
x 0.7586799647899636
y 0.5221511083334797
aaa:string wye:string
b:string wye:string
i:string 3:string
x:string 0.20460330576630303:string
y:string 0.33831852551664776:string
a:string eks:string
bbb:string wye:string
i:string 4:string
x:string 0.38139939387114097:string

aaa wye
b   wye
i   3
x   0.20460330576630303
y   0.33831852551664776
y:string 0.13418874328430463:string

a   eks
bbb wye
i   4
x   0.38139939387114097
y   0.13418874328430463
a:string wye:string
b:string pan:string
i:string 5:string
xxx:string 0.5732889198020006:string
y:string 0.8636244699032729:string
a:string zee:string

a   wye
b   pan
i   5
xxx 0.5732889198020006
y   0.8636244699032729
b:string pan:string
i:string 6:string
x:string 0.5271261600918548:string
y:string 0.49322128674835697:string

a zee
b pan
i 6
x 0.5271261600918548
y 0.49322128674835697
a:string eks:string
b:string zee:string
iii:string 7:string
x:string 0.6117840605678454:string
y:string 0.1878849191181694:string

a   eks
b   zee
iii 7
x   0.6117840605678454
y   0.1878849191181694
a:string zee:string
b:string wye:string
i:string 8:string
x:string 0.5985540091064224:string
yyy:string 0.976181385699006:string
aaa:string hat:string
bbb:string wye:string
i:string 9:string

a   zee
b   wye
i   8
x   0.5985540091064224
yyy 0.976181385699006
x:string 0.03144187646093577:string
y:string 0.7495507603507059:string

aaa hat
bbb wye
i   9
x   0.03144187646093577
y   0.7495507603507059
a:string pan:string
b:string wye:string
i:string 10:string
x:string 0.5026260055412137:string
y:string 0.9526183602969864:string

a pan
b wye
i 10
x 0.5026260055412137
y 0.9526183602969864

mlr --oxtab --from ./reg-test/input/abixy-het put s = joink({1:2, "abc":4, 5:"xyz"}, ":"); $* = splitnv(s, ":")
1 1
2 abc
3 5

1 1
2 abc
3 5

1 1
2 abc
3 5

1 1
2 abc
3 5

1 1
2 abc
3 5

1 1
2 abc
3 5

1 1
2 abc
3 5

1 1
2 abc
3 5

1 1
2 abc
3 5

1 1
2 abc
3 5

mlr --oxtab --from ./reg-test/input/abixy-het put s = joinv({1:2, "abc":4, 5:"xyz"}, ":"); $* = splitnv(s, ":")
1 2
2 4
3 xyz

1 2
2 4
3 xyz

1 2
2 4
3 xyz

1 2
2 4
3 xyz

1 2
2 4
3 xyz

1 2
2 4
3 xyz

1 2
2 4
3 xyz

1 2
2 4
3 xyz

1 2
2 4
3 xyz

1 2
2 4
3 xyz

mlr --oxtab --from ./reg-test/input/abixy-het put s = joinkv({1:2, "abc":4, 5:"xyz"}, ":", ";"); $* = splitkv(s, ":", ";"); for (k,v in $*) { print k.":".typeof(k)." ".v.":".typeof(v)}
1:string 2:int
abc:string 4:int
5:string xyz:string
1   2
abc 4
5   xyz
1:string 2:int
abc:string 4:int
5:string xyz:string
1:string 2:int

1   2
abc 4
5   xyz
abc:string 4:int
5:string xyz:string

1   2
abc 4
5   xyz
1:string 2:int
abc:string 4:int
5:string xyz:string
1:string 2:int
abc:string 4:int
5:string xyz:string

1   2
abc 4
5   xyz

1   2
abc 4
5   xyz
1:string 2:int
abc:string 4:int
5:string xyz:string

1   2
abc 4
5   xyz
1:string 2:int
abc:string 4:int
5:string xyz:string

1   2
abc 4
5   xyz
1:string 2:int
abc:string 4:int
5:string xyz:string

1   2
abc 4
5   xyz
1:string 2:int
abc:string 4:int
5:string xyz:string

1   2
abc 4
5   xyz
1:string 2:int
abc:string 4:int
5:string xyz:string

1   2
abc 4
5   xyz

mlr --oxtab --from ./reg-test/input/abixy-het put s = joinkv({1:2, "abc":4, 5:"xyz"}, ":", ";"); $* = splitkvx(s, ":", ";"); for (k,v in $*) { print k.":".typeof(k)." ".v.":".typeof(v)}
1:string 2:string
abc:string 4:string
5:string xyz:string
1   2
abc 4
5   xyz
1:string 2:string
abc:string 4:string
5:string xyz:string
1:string 2:string
abc:string 4:string
5:string xyz:string

1   2
abc 4
5   xyz

1   2
abc 4
5   xyz
1:string 2:string
abc:string 4:string
5:string xyz:string

1   2
abc 4
5   xyz
1:string 2:string
abc:string 4:string
5:string xyz:string

1   2
abc 4
5   xyz
1:string 2:string
abc:string 4:string
5:string xyz:string

1   2
abc 4
5   xyz
1:string 2:string
abc:string 4:string
5:string xyz:string

1   2
abc 4
5   xyz
1:string 2:string
abc:string 4:string
5:string xyz:string

1   2
abc 4
5   xyz
1:string 2:string
abc:string 4:string
5:string xyz:string
1:string 2:string

1   2
abc 4
5   xyz
abc:string 4:string
5:string xyz:string

1   2
abc 4
5   xyz

